### ISSUE TITLE
Error in `Lint/RedundantCopDisableDirective`

### ISSUE THREAD

**Issue by Tietew** [rubocop/rubocop#11160]
An error occurred while Lint/RedundantCopDisableDirective cop was inspecting following file:

``` ruby
class Foo
  # rubocop:disable Style
  include Bar
  # rubocop:enable all
end
```

--------

## Expected behavior

No errors.

## Actual behavior

```console
$ be rubocop test.rb -d
For ...: configuration from .../.rubocop.yml
Default configuration from .../gems/rubocop-1.38.0/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning .../test.rb
An error occurred while Lint/RedundantCopDisableDirective cop was inspecting .../test.rb.
Couldn't find Style/ArrayCoercion in comment: # rubocop:disable Style
.../gems/rubocop-1.38.0/lib/rubocop/cop/lint/redundant_cop_disable_directive.rb:256:in `cop_range'
.../gems/rubocop-1.38.0/lib/rubocop/cop/lint/redundant_cop_disable_directive.rb:233:in `block in add_offense_for_some_cops'
.../lib/ruby/3.1.0/set.rb:511:in `each_key'
.../lib/ruby/3.1.0/set.rb:511:in `each'
.../gems/rubocop-1.38.0/lib/rubocop/cop/lint/redundant_cop_disable_directive.rb:233:in `map'
.../gems/rubocop-1.38.0/lib/rubocop/cop/lint/redundant_cop_disable_directive.rb:233:in `add_offense_for_some_cops'
.../gems/rubocop-1.38.0/lib/rubocop/cop/lint/redundant_cop_disable_directive.rb:212:in `block in add_offenses'
.../gems/rubocop-1.38.0/lib/rubocop/cop/lint/redundant_cop_disable_directive.rb:208:in `each'
.../gems/rubocop-1.38.0/lib/rubocop/cop/lint/redundant_cop_disable_directive.rb:208:in `add_offenses'
.../gems/rubocop-1.38.0/lib/rubocop/cop/lint/redundant_cop_disable_directive.rb:51:in `on_new_investigation'
.../gems/rubocop-1.38.0/lib/rubocop/cop/commissioner.rb:153:in `block (2 levels) in invoke'
.../gems/rubocop-1.38.0/lib/rubocop/cop/commissioner.rb:160:in `with_cop_error_handling'
.../gems/rubocop-1.38.0/lib/rubocop/cop/commissioner.rb:153:in `block in invoke'
.../gems/rubocop-1.38.0/lib/rubocop/cop/commissioner.rb:153:in `each'
.../gems/rubocop-1.38.0/lib/rubocop/cop/commissioner.rb:153:in `invoke'
.../gems/rubocop-1.38.0/lib/rubocop/cop/commissioner.rb:84:in `investigate'
.../gems/rubocop-1.38.0/lib/rubocop/cop/team.rb:154:in `investigate_partial'
.../gems/rubocop-1.38.0/lib/rubocop/cop/team.rb:82:in `investigate'
.../gems/rubocop-1.38.0/lib/rubocop/runner.rb:315:in `inspect_file'
.../gems/rubocop-1.38.0/lib/rubocop/runner.rb:172:in `block in add_redundant_disables'
.../gems/rubocop-1.38.0/lib/rubocop/runner.rb:193:in `team_for_redundant_disables'
.../gems/rubocop-1.38.0/lib/rubocop/runner.rb:171:in `add_redundant_disables'
.../gems/rubocop-1.38.0/lib/rubocop/runner.rb:139:in `block in file_offenses'
.../gems/rubocop-1.38.0/lib/rubocop/runner.rb:163:in `file_offense_cache'
.../gems/rubocop-1.38.0/lib/rubocop/runner.rb:137:in `file_offenses'
.../gems/rubocop-1.38.0/lib/rubocop/runner.rb:128:in `process_file'
.../gems/rubocop-1.38.0/lib/rubocop/runner.rb:109:in `block in each_inspected_file'
.../gems/rubocop-1.38.0/lib/rubocop/runner.rb:108:in `each'
.../gems/rubocop-1.38.0/lib/rubocop/runner.rb:108:in `reduce'
.../gems/rubocop-1.38.0/lib/rubocop/runner.rb:108:in `each_inspected_file'
.../gems/rubocop-1.38.0/lib/rubocop/runner.rb:94:in `inspect_files'
.../gems/rubocop-1.38.0/lib/rubocop/runner.rb:47:in `run'
.../gems/rubocop-1.38.0/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
.../gems/rubocop-1.38.0/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
.../gems/rubocop-1.38.0/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
.../gems/rubocop-1.38.0/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
.../gems/rubocop-1.38.0/lib/rubocop/cli/command.rb:11:in `run'
.../gems/rubocop-1.38.0/lib/rubocop/cli/environment.rb:18:in `run'
.../gems/rubocop-1.38.0/lib/rubocop/cli.rb:72:in `run_command'
.../gems/rubocop-1.38.0/lib/rubocop/cli.rb:79:in `execute_runners'
.../gems/rubocop-1.38.0/lib/rubocop/cli.rb:48:in `run'
.../gems/rubocop-1.38.0/exe/rubocop:19:in `block in <top (required)>'
.../lib/ruby/3.1.0/benchmark.rb:311:in `realtime'
.../gems/rubocop-1.38.0/exe/rubocop:19:in `<top (required)>'
.../bin/rubocop:25:in `load'
.../bin/rubocop:25:in `<top (required)>'
.../lib/ruby/gems/3.1.0/gems/bundler-2.3.14/lib/bundler/cli/exec.rb:58:in `load'
.../lib/ruby/gems/3.1.0/gems/bundler-2.3.14/lib/bundler/cli/exec.rb:58:in `kernel_load'
.../lib/ruby/gems/3.1.0/gems/bundler-2.3.14/lib/bundler/cli/exec.rb:23:in `run'
.../lib/ruby/gems/3.1.0/gems/bundler-2.3.14/lib/bundler/cli.rb:483:in `exec'
.../lib/ruby/gems/3.1.0/gems/bundler-2.3.14/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
.../lib/ruby/gems/3.1.0/gems/bundler-2.3.14/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
.../lib/ruby/gems/3.1.0/gems/bundler-2.3.14/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
.../lib/ruby/gems/3.1.0/gems/bundler-2.3.14/lib/bundler/cli.rb:31:in `dispatch'
.../lib/ruby/gems/3.1.0/gems/bundler-2.3.14/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
.../lib/ruby/gems/3.1.0/gems/bundler-2.3.14/lib/bundler/cli.rb:25:in `start'
.../lib/ruby/gems/3.1.0/gems/bundler-2.3.14/exe/bundle:48:in `block in <top (required)>'
.../lib/ruby/gems/3.1.0/gems/bundler-2.3.14/lib/bundler/friendly_errors.rb:103:in `with_friendly_errors'
.../lib/ruby/gems/3.1.0/gems/bundler-2.3.14/exe/bundle:36:in `<top (required)>'
.../bin/bundle:25:in `load'
.../bin/bundle:25:in `<main>'
.
```

## RuboCop version

```
$ [bundle exec] rubocop -V
1.38.0 (using Parser 3.1.2.1, rubocop-ast 1.23.0, running on ruby 3.1.2) [x86_64-linux]
```

**Comment by koic** [comment]
This issue may be the same as https://github.com/rubocop/rubocop/issues/11143.

**Comment by nevans** [comment]
For what it's worth, I've also experienced this exact same bug, with a different Style cop, but with the exact same stacktrace.

I can trigger it with the following file:
```ruby
# rubocop:disable Style
def foo bar

end
# rubocop:enable Style
``` 

And in a yaml file:
```yaml
Style/AccessorGrouping: { Enabled: false }
```

And then run with `rubocop -c tmp/rubocop-err.yml tmp/rubocop-err.rb

With some experimentation, the "Couldn't find" comment seems to reference the first (alphabetically) `Style` cop that with `Enabled: false` in the config file.  If I enable the cop with the error, the next cop (alphabetically) that is disabled will cause the error.

**Comment by koic** [comment]
I'm going close this issue because it has been resolved with latest RuboCop 1.41. Please upgrade to the latest RuboCop.
